### PR TITLE
JAMES-2378 Log ValidationResult upon JMAP message update validation fail

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java [hunk 1]
```diff
@@ -320,7 +320,7 @@ private void addMessageIdNotFoundToResponse(MessageId messageId, SetMessagesResp
 
     private void handleInvalidRequest(SetMessagesResponse.Builder responseBuilder, MessageId messageId,
                                       List<ValidationResult> validationErrors) {
-        LOGGER.error("Invalid update request for message #{}", messageId);
+        LOGGER.warn("Invalid update request for message #{}: {}", messageId, validationErrors);
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java]: not sure putting a list in a log will work, every List implementation don't have a good toString.
We should write `validationsErrors.stream().collect(Collectors.join(", "))` or something like that.

Comment (chibenwa) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesUpdateProcessor.java]: Only immutable lists are passed to that method. Immutable lists have good rendering. I would be in favor of enforcing an immutable list as argument.

**Comment by chibenwa** [comment]
Sorry, I failed the merge (seems weird). Here is the diff forced pushed to end up with the same state...

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**